pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/offline/AbstractTableDataManager.java

```diff
@@ -30,78 +33,79 @@
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
 import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 
-// TODO: pass a reference to Helix property store during initialization. Both OFFLINE and REALTIME use case need it.
 public abstract class AbstractTableDataManager implements TableDataManager {
-  protected final List<String> _activeSegments = new ArrayList<String>();
-  protected final List<String> _loadingSegments = new ArrayList<String>();
+  protected final List<String> _activeSegments = new ArrayList<>();
+  protected final List<String> _loadingSegments = new ArrayList<>();
   // This read-write lock protects the _segmentsMap and SegmentDataManager.refCnt
   protected final ReadWriteLock _rwLock = new ReentrantReadWriteLock();
   @VisibleForTesting
-  protected final Map<String, SegmentDataManager> _segmentsMap = new HashMap<String, SegmentDataManager>();
+  protected final Map<String, SegmentDataManager> _segmentsMap = new HashMap<>();
 
-  protected Logger LOGGER = LoggerFactory.getLogger(AbstractTableDataManager.class);
-  protected volatile boolean _isStarted = false;
-  protected String _tableName;
+  // Logger should be set up in doInit()
+  protected Logger _logger;
```

Comment:
You should set it to the default logger, to avoid NPE just in case someone adds code to log before calling doInit()

Reply:
Fixed by setting up the logger in init()